h/streamer/websocket.py

```diff
@@ -18,8 +18,28 @@
 # below.
 MESSAGE_HANDLERS = {}
 
+
 # An incoming message from a WebSocket client.
-Message = namedtuple('Message', ['socket', 'payload'])
+class Message(namedtuple('Message', [
+    'socket',
+    'payload',
+])):
+    def reply(self, payload, ok=True):
+        """
+        Send a response to this message.
+
+        Sends a reply message back to the client, with the passed `payload`
+        and reporting status `ok`.
+        """
+        reply_to = self.payload.get('id')
+        # Short-circuit if message is missing an ID or has a non-numeric ID.
+        if not isinstance(reply_to, (int, float)):
+            return
```

Comment:
Why would that be nicer? That means that the responsibility for extracting and validating the ID has to be repeated in each and every message handler, no? That seems to me to be *more* likely to introduce bugs, and thus "situations where it's hard to debug why the client isn't getting a reply"...

As for whether we'd immediately close if we got a message without an ID... I don't think so. It's potentially a nice way for the client to signal it doesn't care about the response.

For now the rule is simple: if you send a numeric ID, you get a response. If you don't, you don't.